### ISSUE TITLE
Unresolved import for cross-compiled project

### ISSUE THREAD

**Issue by donotnoot** [rust-analyzer/rust-analyzer#7120]
Hey everyone. First off I'm really sorry if this is a dupe of #6038. From what I can gather reading the comments on that issue it looks like the problem there is expanding the `cfg_if` and the issue I'm running into does not have any `cfg_if` involved, therefore I'm not 100% sure whether it is the same thing. I'm quite a Rust noob however so it could very well be that I'm missing something and it's actually the exact same thing.

As for the report, I'm working on an embedded project using RTIC. RA is reporting that a module is not a resolved import even though the code compiles and works fine, I can see the pin switching state in an oscilloscope.

Reproducing this is rather easy, there is a blinky example already on github that is very small and manages to trigger this behaviour so I'll just use that:

```
/tmp/tmp.IrRGT4EEti % git clone https://github.com/mitchmindtree/stm32f4xx-rtic-blinky
[blah blah, cloned it]
/tmp/tmp.IrRGT4EEti % cd stm32f4xx-rtic-blinky
/tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky master % cargo build --release
[blah blah, compiled a bunch of stuff]
    Finished release [optimized + debuginfo] target(s) in 55.25s
/tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky master % rust-analyzer --version
rust-analyzer 1d53075
/tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky master % rust-analyzer diagnostics .
processing crate: stm32f4xx_rtic_blinky, module: /tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky/build.rs
processing crate: stm32f4xx_rtic_blinky, module: /tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky/src/main.rs
Diagnostic { message: "unresolved import", range: 130..150, severity: Error, fix: None, unused: false, code: Some(DiagnosticCode("unresolved-import")) }

diagnostic scan complete

diagnostic error detected
```

I know that RTIC uses loads of macros so I tried with the `proc-macro` option too:
```
/tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky master % rust-analyzer -v diagnostics --with-proc-macro .
processing crate: stm32f4xx_rtic_blinky, module: /tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky/build.rs
processing crate: stm32f4xx_rtic_blinky, module: /tmp/tmp.IrRGT4EEti/stm32f4xx-rtic-blinky/src/main.rs
Diagnostic { message: "unresolved import", range: 130..150, severity: Error, fix: None, unused: false, code: Some(DiagnosticCode("unresolved-import")) }

diagnostic scan complete

diagnostic error detected
```

Also, if I open that with my text editor I get the following complaints from RA:
- As expected, line 8 of main.rs is flagged with the unresolved import
- If I go into the RTIC crate, I get a warning on [this line](https://github.com/rtic-rs/cortex-m-rtic/blob/e65a86f6962218b66a14b7ccc40eec7976880eba/src/lib.rs#L47) that "code is inactive due to #[cfg] directives: armv7m is disabled", which is why R-A thinks that the module is not imported

Once again sorry if this is a dupe.

**Comment by flodiebold** [comment]
Does setting `rust-analyzer.cargo.target` to `thumbv7em-none-eabihf` in the workspace settings help?

**Comment by donotnoot** [comment]
Hi @flodiebold thanks for getting back. Apologies, I forgot to mention that I did try that.

I did it again quickly with VS Code just to make sure it wasn't Neovim not properly setting the config. I used the following config file for VS Code:
```{
    "workbench.colorTheme": "Visual Studio Light",
    "editor.minimap.enabled": false,
    "rust-analyzer.cargo.target": "thumbv7em-none-eabihf"
}```

Unfortunately it does not seem to help, the exact same behavior occurs.

**Comment by jonas-schievink** [comment]
Please enable "Load out dirs from check", that should fix it

**Comment by donotnoot** [comment]
@jonas-schievink yeah, that fixed it. Thanks! feel free to close.
